### PR TITLE
add functionality to get articles filtered by topics, sorted by creat…

### PR DESCRIPTION
**Pull request by AlexMSm**
…ed_at date descending. Also removed a rogue controller I accidentally created for the last task that wasn't needed.

### CODE REVIEW

File: __tests__/app.articles.test.js [hunk 1]
```diff
@@ -152,4 +152,68 @@ describe("/api/articles", () => {
         });
     });
   });
+  describe("GET /api/articles?topic=<topic> - returns array of article objects with optional topic filter", () => {
+    test("200: returns array of all article objects DATE DSC", () => {
```
Comment (MrMcSnail) [inline __tests__/app.articles.test.js:156]: You might want to look at [Jest Sorted](https://www.npmjs.com/package/jest-sorted) for this and similar situations

File: app.js [hunk 2]
```diff
@@ -7,14 +7,17 @@ const {
...
 
 const { getUsers } = require("./_controllers/users.controllers");
 
+app.get("/api/articles", getArticlesByTopic);
+
 app.get("/api/topics", getTopics);
 app.get("/api/articles/:article_id", getArticleById);
 app.get("/api/users", getUsers);
-app.get("/api/articles/:article_id", getArticleCommentCount);
+//app.get("/api/articles/:article_id", getArticleCommentCount);
```
Comment (AlexMSm) [inline app.js]: This will be deleted with this merge.

File: __tests__/app.articles.test.js [hunk 3]
```diff
@@ -152,4 +152,68 @@ describe("/api/articles", () => {
...
+        });
+    });
+    test("400 - Bad request: returns error when queried topic does not exists", () => {
+      return request(app)
+        .get("/api/articles?topic=barnacles")
+        .expect(400)
+        .then((response) => {
+          const { body } = response;
+          expect(body.msg).toBe(
+            "barnacles is not a valid topic - available topics: mitch,cats,paper"
```
Comment (MrMcSnail) [inline __tests__/app.articles.test.js:214]: Nice bit of dynamic error messaging

File: app.js [hunk 4]
```diff
@@ -27,6 +30,7 @@ app.use((err, req, res, next) => {
 });
 
 app.use((err, req, res, next) => {
+  console.log(err);
```
Comment (MrMcSnail) [inline app.js:32]: You might want to still have an app.all error handler to make sure you can handle 404 errors that are in the routing and outwith the scope of your controllers and models

**Review by MrMcSnail — APPROVED** [review]
Good to merge after reading comments

**Pull request merged**